modules/regex/regex_mod.c

```diff
@@ -249,9 +257,9 @@ static int mod_init(void)
 
 	return 0;
 
-err:
-	free_shared_memory();
-	return -1;
+	err:
```

Comment:
Please revert this change, it is misleading.

Reply:
done.